commit-event-service/src/test/scala/io/renku/commiteventservice/events/categories/globalcommitsync/eventgeneration/gitlab/GitLabCommitFetcherSpec.scala

```diff
@@ -20,156 +20,250 @@ package io.renku.commiteventservice.events.categories.globalcommitsync.eventgene
       val personalAccessToken = personalAccessTokens.generateOne
 
-      stubFor {
-        get(s"/api/v4/projects/$projectId/repository/commits?page=${pageRequest.page}&per_page=${pageRequest.perPage}")
-          .withHeader("PRIVATE-TOKEN", equalTo(personalAccessToken.value))
-          .willReturn(okJson(commitsJson(from = commitInfoList)))
-      }
+      val expectation = pageResults().generateOne
+
+      (gitLabClient
+        .send(_: Method, _: Uri, _: String Refined NonEmpty)(_: ResponseMappingF[IO, PageResult])(
+          _: Option[AccessToken]
+        ))
+        .expects(GET, uri, endpointName, *, Some(personalAccessToken))
+        .returning(expectation.pure[IO])
 
       gitLabCommitFetcher
         .fetchGitLabCommits(projectId, pageRequest)(Some(personalAccessToken))
-        .unsafeRunSync() shouldBe PageResult(commitInfoList.map(_.id), maybeNextPage = None)
+        .unsafeRunSync() shouldBe expectation
     }
 
     "fetch commits using the given oauth token" in new TestCase {
       val authAccessToken = oauthAccessTokens.generateOne
 
-      stubFor {
-        get(s"/api/v4/projects/$projectId/repository/commits?page=${pageRequest.page}&per_page=${pageRequest.perPage}")
-          .withHeader("Authorization", equalTo(s"Bearer ${authAccessToken.value}"))
-          .willReturn(okJson(commitsJson(from = commitInfoList)))
-      }
+      val expectation = pageResults().generateOne
+
+      (gitLabClient
+        .send(_: Method, _: Uri, _: String Refined NonEmpty)(_: ResponseMappingF[IO, PageResult])(
+          _: Option[AccessToken]
+        ))
+        .expects(GET, uri, endpointName, *, authAccessToken.some)
+        .returning(expectation.pure[IO])
 
       gitLabCommitFetcher
         .fetchGitLabCommits(projectId, pageRequest)(Some(authAccessToken))
-        .unsafeRunSync() shouldBe PageResult(commitInfoList.map(_.id), maybeNextPage = None)
+        .unsafeRunSync() shouldBe expectation
     }
 
     "return no commits if there aren't any" in new TestCase {
 
-      stubFor {
-        get(s"/api/v4/projects/$projectId/repository/commits?page=${pageRequest.page}&per_page=${pageRequest.perPage}")
-          .willReturn(okJson("[]"))
-      }
+      val expectation = PageResult(commits = Nil, maybeNextPage = None)
+
+      (gitLabClient
+        .send(_: Method, _: Uri, _: String Refined NonEmpty)(_: ResponseMappingF[IO, PageResult])(
+          _: Option[AccessToken]
+        ))
+        .expects(GET, uri, endpointName, *, maybeAccessToken)
+        .returning(expectation.pure[IO])
 
       gitLabCommitFetcher
-        .fetchGitLabCommits(projectId, pageRequest)(maybeAccessToken = None)
-        .unsafeRunSync() shouldBe PageResult(commits = Nil, maybeNextPage = None)
+        .fetchGitLabCommits(projectId, pageRequest)(maybeAccessToken)
+        .unsafeRunSync() shouldBe expectation
     }
 
-    "return an empty list if project for NOT_FOUND" in new TestCase {
+    /////////////////// ResponseMapper tests
```

Comment:
Do you find this comment helpful?

Reply:
It was helpful during development but I can remove it now.